src/hotspot/os/linux/os_linux.cpp

```diff
@@ -379,11 +377,11 @@ void os::Linux::kernel_version(long* major, long* minor) {
   }
 
   char* walker = buffer.release;
-  long* set_v = major;
-  while (*minor == -1 && walker != nullptr) {
+  long* set_v = &_release_major;
+  while (_release_minor == -1 && walker != nullptr) {
     if (isdigit(walker[0])) {
       *set_v = strtol(walker, &walker, 10);
-      set_v = minor;
+      set_v = &_release_minor;
     } else {
       ++walker;
     }
```

Comment:
I can't tell if this code is correct or not.
* When is walker ever nullptr?
* Don't wee need error handling for strtol?

Maybe this is all pre-existing issues, but I'd like to get someone else to review and sign off on this code.

Reply:
Not typical code for me either.

Quoting my man pages:

```
     If endptr is  not NULL, strtol() stores the address  of the first
     invalid character  in *endptr.  If  there were no digits  at all,
     strtol()  stores  the original  value  of  nptr in  *endptr  (and
     returns  0).  In  particular, if  *nptr is  not \[aq]\0\[aq]  but
     **endptr is \[aq]\0\[aq] on return, the entire string is valid.
```

Given a string such as `2.6.5`, each `strtol` will put `walker` at a new `.` and as minor is always set the condition in the loop will become false. The `walker` will never be null. Looking at the documentation, we don't need error-handling for the `strtol` call (won't over/underflow and will be a valid version revision).

Reply:
Basic fix: change `walker != nullptr` to `walker[0] != '\0'` and add a test case for this.